src/org/traccar/protocol/Arnavi4FrameDecoder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 Ivan Muratov (binakot@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.channel.Channel;
+import org.jboss.netty.channel.ChannelHandlerContext;
+import org.jboss.netty.handler.codec.frame.FrameDecoder;
+
+public class Arnavi4FrameDecoder extends FrameDecoder {
```

Comment:
1. Yes, the first packet is always header. I will add the flag for that.
2. There is no length value in beginning of the packet. We never know where is the real end of packet. We only know that in the end must be end sign byte 0x5D. But I can write some smart check for that. Working on it :octocat: